README.md

```diff
@@ -42,12 +42,15 @@ Future features include managing/creating email campaigns and ticket exchanges.
 3. Create mkcert certificates.
    1. Navigate to `<path/to/WonderTix/server>` and `<path/to/WonderTix/client>`.
    2. Run `mkcert -install` to install the local certificate authority.
-   3. Run `mkcert localhost` to create a certificate.   
+   3. Run `mkcert localhost` to create a certificate.
+   4. Node does not automatically look in the system root store for certificate authority files, nor would Docker include it anyways. Thus, the certificate authority generated by mkcert must be copied to a location where the server can find it so features like WebSockets function. To do so, in the server directory, run the following command:
+      - On Windows Powershell: `copy ($(mkcert -CAROOT) + '/rootCA.pem') .`.
+      - On Mac/Linux: `cp $(mkcert -CAROOT)/rootCA.pem .`.
 4. Run `docker-compose up -d` (or `docker compose up` to see live container logs in the terminal). This starts a locally running instance of Wondertix using docker containers. 
 5. To test the checkout process with Stripe, make sure you have the [Stripe CLI](https://stripe.com/docs/stripe-cli) installed. 
    1. Once the Stripe CLI is installed, run `stripe login` and press enter to open the browser. 
    2. In the browser that the Stripe CLI opens, make sure you select `Wondertix Dev` as the account, verify that the code matches that in your terminal, and select `Allow Access`. If `Wondertix Dev` does not appear as an option, you will need to ask for access to the account. You should only need to login to the account like this once. 
-   3. Back in your terminal, run `stripe listen --forward-to https://localhost:8000/api/2/order/webhook --events checkout.session.completed,checkout.session.expired` and copy the resulting ***signing secret*** as your `PRIVATE_STRIPE_WEBHOOK` variable in the `.env` file. This tells Stripe to forward events to your local webhook. See the `stripe listen` command [API docs](https://stripe.com/docs/cli/listen) for more details. 
+   3. Back in your terminal, run `stripe listen --forward-to https://localhost:8000/api/2/order/webhook` and copy the resulting ***signing secret*** as your `PRIVATE_STRIPE_WEBHOOK` variable in the `.env` file. This tells Stripe to forward events to your local webhook. See the `stripe listen` command [API docs](https://stripe.com/docs/cli/listen) for more details. 
```

Comment:
Having not looked into it, is there a way to complete a reader checkout locally (eg. provide a payment_method to stripe)? Follow up, do you know if there would there be a way to test a complete reader checkout in a deployed environment? Would you mind adding instructions for doing either/or here?

Reply:
> Having not looked into it, is there a way to complete a reader checkout locally (eg. provide a payment_method to stripe)? Follow up, do you know if there would there be a way to test a complete reader checkout in a deployed environment? Would you mind adding instructions for doing either/or here?

Reader checkouts can be completed locally. I assume the same method for completing them locally would work in a deployed environment. The instructions are as follows:

1. Ensure you have the Stripe CLI installed.
2. Turn on the servers and ensure that the webhook is listening to Stripe events. These events are diff
3. First, set up the simulated reader. It is possible to this via the Stripe CLI or through the Stripe API, but these instructions will explain how to do so through the Stripe dashboard as I guess that is how readers will often be set up in practice.
a. Go to https://dashboard.stripe.com/test/terminal. 
b. Add a location if one does not already exist.
c. Click on the location in the location list.
d. Press register reader.
e. Use `simulated-wpe` as the registration code.
f. Once created, click on the reader in the list to get an info pop up to appear. Write down the Reader ID somewhere, which should have the format `tmr_xxx`.
4. Navigate to the deployed front end (ex: `https://localhost:3000`). Log in, and go to the admin ticketing page.
5. Create your sample purchase. Once it is ready, press the "send to reader" button. The adjacent drop down menu should display a list of possible readers to send to; as you have only set up one reader, the only option should match the reader ID noted earlier.
6. You should be taken to the reader checkout page.
a. It is worth noting that here you can cancel the payment early *if a payment method has not yet been presented*. It also can directly poll the Stripe API for cases when the webhook is slow to confirm a payment, or if the webhook is not listening to events for whatever reason.
7. Open a terminal.
8. Run this command: `stripe test_helpers terminal readers present_payment_method tmr_xxx`, replacing `tmr_xxx` with your Reader ID.
9. The frontend page should soon redirect you to the success page.

These instructions were based on the premise you are testing with a simulated reader and a virtual card. Based on what I read, they should not vary much if you were instead working with a physical reader and a physical test card. For more information on how to set up either simulated readers or real readers, [see here](https://docs.stripe.com/terminal/payments/connect-reader?reader-type=simulated). For more information on simulated card presenting, [see here](https://docs.stripe.com/terminal/references/testing?lang=cli#simulated-card-presentment). For physical test cards, [see here](https://docs.stripe.com/terminal/references/testing?lang=cli#physical-test-cards).

I will add a simplified form of these instructions to the readme shortly.

Reply:
Also, if you want to test the error popups for declined payments, you can use one of [these](https://docs.stripe.com/testing) test numbers that simulate failures, for example to simulate an insufficient funds error, you would run `stripe test_helpers terminal readers present_payment_method tmr_xxx --card_present.number 4000000000009995`

Reply:
I got these steps to work except for the `stripe test_helpers ...` command. I had to access to API endpoint directly, not sure if stripe was configured with the sk or not
```
curl -X POST https://api.stripe.com/v1/test_helpers/terminal/readers/tmr_.../present_payment_method \
  -u "sk_test_...:"
```

Reply:
if you can only use curl, then to supply a card number i believe you would do `curl -X POST https://api.stripe.com/v1/test_helpers/terminal/readers/tmr_xxx/present_payment_method -u "sk_test_xxx:" -d "type=card_present" -d "card_present[number]=4000000000009995"`